File: backend/src/main/java/com/back/team11/domain/cafe/batch/dto/KakaoSearchResponse.java
package com.back.team11.domain.cafe.batch.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

@JsonIgnoreProperties(ignoreUnknown = true)
public record KakaoSearchResponse(
        List<KakaoPlaceDto> documents,
        Meta meta
) {
    public record Meta(
            @JsonProperty("is_end") boolean isEnd
    ){}
}

File: backend/src/main/java/com/back/team11/domain/cafe/batch/dto/KakaoSearchResponse.kt
package com.back.team11.domain.cafe.batch.dto

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

@JsonIgnoreProperties(ignoreUnknown = true)
data class KakaoSearchResponse(
    val documents: List<KakaoPlaceDto>,
    val meta: Meta,
) {
    data class Meta(
        @JsonProperty("is_end") val isEnd: Boolean,
    )
}